source/level.js

```diff
@@ -7,29 +7,46 @@ export class Level extends Phaser.Scene{
     // General
 	width = 0;
 	height = 0;
-    background;
-    minimap;
+    background = {
+        name: '',
+        path: ''
+    }
+    minimap = {
+        name: '',
+        path: ''
+    }
     timeDelta = 0;
     // Character
     characterData = {
         gold: 0,
         characterClass: '',
-        enemiesKilled: 0,
+        totalEnemiesKilled: 0,
         timesClicked: 0,
         damageByClicking: 0,
         damageByAutoClick: 0,
         numberOfAutoClickers: 0,
-        unlockedLevels: {
-        	tutorialIsland: true,
-        	lumbridge: true,
-        	varrock: false
+        // Can be accessed with characterData[this.background.name].questCompleted, etc.
+        tutorialIsland: {
+            questCompleted: true,
+            enemiesKilled: 0
+        },
+        lumbridge: {
+            questCompleted: false,
+            enemiesKilled: 0
+        },
+        varrock: {
+            questCompleted: false,
+            enemiesKilled: 0
         }
     };
     // Autoclickers
     autoClickers = [];
     autoClickDps = 0;
     // Enemy
-	enemySettings; // From constructor
+	enemySettings = { // From constructor
```

Comment:
weird tab here, just look through file again for weird tabbing - it happens elsewhere